libs/browser/src/device.rs

```diff
@@ -0,0 +1,222 @@
+use std::{
+    str::FromStr,
+    sync::{
+        atomic::{AtomicBool, AtomicUsize, Ordering},
+        Arc, RwLock,
+    },
+};
+
+use gtk::prelude::*;
+use webkit2gtk::{WebInspectorExt, WebView, WebViewExt};
+
+/// Configuration of a VX device.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Configuration {
+    pub name: &'static str,
+    pub dimensions: (i32, i32),
```

Comment:
I also have set in my current device emulation configs two other settings: device pixel ratio and user agent. Are either of those relevant here?
<img width="359" alt="Screenshot 2024-12-03 at 3 15 03 PM" src="https://github.com/user-attachments/assets/821d5ea5-9736-47c4-b63a-e95118096b9f">

Reply:
I'm not sure exactly what the device pixel ratio does. Is it just changing the scaling factor of the webview? I'm not sure how to do that, but we may be able to add something to that effect. We don't actually care about user agent, right?

Reply:
I'm also not sure about what device pixel ratio does. I feel like @kofi-q might know.

I don't know that user agent matters. In Chrome, it will change the cursor to a touch circle, which is minorly useful in making sure touch targets are big enough and remember that it's a touch screen app. It may also change how clicks are translated into browser events. Neither of those seem like things we rely on heavily, but I might be missing something.

Reply:
Ah, good points. @kofi-q, mind taking a little time to try this out? I'm not sure what is actually important vs a nice to have for UI development in kiosk-browser.